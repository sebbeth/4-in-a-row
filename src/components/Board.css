
table {
    margin-left: auto;
    margin-right: auto;
}

th,
td {
    padding: 15px;
    text-align: center;
}

.footer {
    display: flex;
    justify-content: space-around;
    align-items: baseline;
    padding: 0.7rem 1.8rem;
    height: 8rem;
}

.footer-text {
    padding:0.5rem;
}

