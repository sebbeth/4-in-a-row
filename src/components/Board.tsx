import React from 'react';
import './Board.css';
import {Column} from './Column';
import { ColumnButton } from './ColumnButton';


export class Board extends React.Component {
    render() {

    const columns = new Array(4).fill(4);
    const rows = new Array(6).fill(6);
    console.log(columns);

    return (
        <div className="board">
        <h1>--Board--</h1>
        {/* <div>
        { columns.map((column) => {
            return (<Column rowCount={rowCount}/>);
        }
        )}
        </div> */}
        <table className="board-table">  
            { rows.map(() => {return (<tr className="row">
                { columns.map(() => {return (<td>cell</td>);})}
            </tr>);})}
            <tr>{ columns.map(() => {return (<th><ColumnButton/></th>);})}</tr>
        </table>
        </div>
    );
    }

    renderColumns() {
        return (<span>COLS</span>);
    }
}