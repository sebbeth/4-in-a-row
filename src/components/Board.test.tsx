import React from 'react';
import ReactDOM from 'react-dom';
import { Board } from './Board';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Board />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('matches the snapshot', () => {
    const component = renderer.create(
        <Board test={true}/>
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot(); 

});

