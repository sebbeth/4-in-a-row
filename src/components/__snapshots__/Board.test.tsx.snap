// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div>
  <h1>
    4 In a Row
  </h1>
  <h2>
    Player One
  </h2>
  <table
    className="board-table"
  >
    <tbody>
      <tr
        className="row"
      >
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
      </tr>
      <tr
        className="row"
      >
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
      </tr>
      <tr
        className="row"
      >
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
      </tr>
      <tr
        className="row"
      >
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
      </tr>
      <tr
        className="row"
      >
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
      </tr>
      <tr
        className="row"
      >
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
      </tr>
      <tr
        className="row"
      >
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
      </tr>
      <tr
        className="row"
      >
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
        <td
          className="empty"
          style={
            Object {
              "backgroundColor": undefined,
            }
          }
        />
      </tr>
      <tr>
        <th>
          <button
            onClick={[Function]}
          >
            Go
          </button>
        </th>
        <th>
          <button
            onClick={[Function]}
          >
            Go
          </button>
        </th>
        <th>
          <button
            onClick={[Function]}
          >
            Go
          </button>
        </th>
        <th>
          <button
            onClick={[Function]}
          >
            Go
          </button>
        </th>
        <th>
          <button
            onClick={[Function]}
          >
            Go
          </button>
        </th>
        <th>
          <button
            onClick={[Function]}
          >
            Go
          </button>
        </th>
        <th>
          <button
            onClick={[Function]}
          >
            Go
          </button>
        </th>
        <th>
          <button
            onClick={[Function]}
          >
            Go
          </button>
        </th>
      </tr>
    </tbody>
  </table>
  <div
    className="footer"
  >
    <span
      className="footer-text"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      Player 1 Color: 
    </span>
    <span
      className="footer-text"
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
    >
      Player 2 Color: 
    </span>
  </div>
</div>
`;
